lessons/Assignment-4-Ruby-Part-2.md

```diff
@@ -1,91 +1,101 @@
 
     ```ruby
     class Book
       attr_reader :author, :title, :count
-      def initialize(author,title,count)
+
+      def initialize(author, title, count)
         @author = author
         @title = title
         @count = count
       end
+
       def to_s
         "author: #{author} title: #{title} count: #{count}"
       end
     end
 
     book_array = []
-    book_array.push(Book.new("Beatrice Potter","Peter Rabbit",25))
-    book_array.push(Book.new("Henry Fielding","Tom Jones",12))
-    book_array.push(Book.new("Bob Woodward","All the President's Men",30))
+    book_array << Book.new("Beatrice Potter", "Peter Rabbit", 25)
+    book_array << Book.new("Henry Fielding", "Tom Jones", 12)
+    book_array << Book.new("Bob Woodward", "All the President's Men", 30)
 
     puts "Sorting alphabetically by author"
 
-    new_array = book_array.sort do |a,b|
-    author1 = a.author.downcase
-    author2 = b.author.downcase
-    author1 <=> author2
-    # if author1 > author2
-    #   1
-    # elsif author1 < author2
-    #   -1
-    # else
-    #   0
-    # end
+    new_array = book_array.sort do |a, b|
+      author1 = a.author.downcase
+      author2 = b.author.downcase
+
+      author1 <=> author2
+      # if author1 > author2
+      #   1
+      # elsif author1 < author2
+      #   -1
+      # else
+      #   0
+      # end
     end
     puts new_array
     ```
 
-    Note the use of the spaceship operator <=>.  It does what the commented out
+    Note the use of the spaceship operator `<=>`.  It does what the commented out
     lines do, but on one line.
 
-    Add to this program. Add additional calls to book_array.sort that pass blocks.  For your first
+    Add to this program. Add additional calls to `book_array.sort` that pass blocks.  For your first
     call to sort, pass a block so that the array is sorted in order of title,
     and print out the array.  For your second call to sort, pass a block so that
     the array is sorted in order of copies, and print out the array.
 
-6. Create a program block_function.rb.  It should have a function do_calc that calls a block using a yield statement.  The yield statement will pass the numbers 7 and 9 to a block, and then do_calc will print out the result returned by the yield.  Call the do_calc function twice in your program. The first time, pass a block that adds the two numbers. The second time, pass a block that multiplies the two numbers.  Your program should print out 16 and 63.
+6. Create a program `block_method.rb`.  It should have a method `do_calc` that calls a block using a `yield` statement.  The yield statement will pass the numbers `7` and `9` to a block, and then `do_calc` will print out the result returned by the yield.  Call the `do_calc` method twice in your program. The first time, pass a block that adds the two numbers. The second time, pass a block that multiplies the two numbers.  Your program should print out `16` and `63`.
```

Comment:
If the rename to `block_method.rb` goes through, we'll have to run `git mv` on the answers.